libraries/chain/witness_schedule.cpp

```diff
@@ -59,36 +65,63 @@ void update_median_witness_props( database& db )
+   int64_t median_available_witness_account_subsidies = active[active.size()/2]->available_witness_account_subsidies;
+
+   rd_system_params account_subsidy_system_params;
+   account_subsidy_system_params.resource_unit = STEEM_ACCOUNT_SUBSIDY_PRECISION;
+   account_subsidy_system_params.decay_per_time_unit_denom_shift = STEEM_RD_DECAY_DENOM_SHIFT;
+   rd_user_params account_subsidy_user_params;
+   account_subsidy_user_params.budget_per_time_unit = median_account_subsidy_budget;
+   account_subsidy_user_params.decay_per_time_unit = median_account_subsidy_decay;
+
+   rd_user_params account_subsidy_per_witness_user_params;
+   int64_t w_budget = median_account_subsidy_budget;
+   w_budget = (w_budget * STEEM_WITNESS_SUBSIDY_BUDGET_PERCENT) / STEEM_100_PERCENT;
+   w_budget = std::min( w_budget, int64_t(std::numeric_limits<int32_t>::max()) );
+   uint64_t w_decay = median_account_subsidy_decay;
+   w_decay = (w_decay * STEEM_WITNESS_SUBSIDY_DECAY_PERCENT) / STEEM_100_PERCENT;
+   w_decay = std::min( w_decay, uint64_t(std::numeric_limits<uint32_t>::max()) );
+
+   account_subsidy_per_witness_user_params.budget_per_time_unit = int32_t(w_budget);
+   account_subsidy_per_witness_user_params.decay_per_time_unit = uint32_t(w_decay);
+
+   // Should never fail, as validate_user_params() is checked and median of valid params should always be valid
+   rd_validate_user_params( account_subsidy_user_params );
 
    db.modify( wso, [&]( witness_schedule_object& _wso )
    {
       _wso.median_props.account_creation_fee       = median_account_creation_fee;
       _wso.median_props.maximum_block_size         = median_maximum_block_size;
       _wso.median_props.sbd_interest_rate          = median_sbd_interest_rate;
-      _wso.median_props.account_subsidy_daily_rate = median_account_subsidy_daily_rate;
-      _wso.median_props.account_subsidy_pool_cap   = median_account_subsidy_pool_cap;
-
-      _wso.account_subsidy_print_rate =
-         ( uint64_t( median_account_subsidy_daily_rate ) * STEEM_ACCOUNT_SUBSIDY_PRECISION ) / STEEM_BLOCKS_PER_DAY;
-      _wso.single_witness_subsidy_limit =
-         (   uint64_t( median_account_subsidy_daily_rate )
-           * uint64_t( median_account_subsidy_pool_cap )
-           * ( uint64_t( STEEM_ACCOUNT_SUBSIDY_PRECISION )
-           *   uint64_t( STEEM_WITNESS_SUBSIDY_PERCENT   ) )
-         ) / STEEM_100_PERCENT;
+      _wso.median_props.account_subsidy_budget     = median_account_subsidy_budget;
+      _wso.median_props.account_subsidy_decay      = median_account_subsidy_decay;
+
+      rd_setup_dynamics_params( account_subsidy_user_params, account_subsidy_system_params, _wso.account_subsidy_rd );
+      rd_setup_dynamics_params( account_subsidy_per_witness_user_params, account_subsidy_system_params, _wso.account_subsidy_witness_rd );
+
+      int64_t median_decay = rd_compute_pool_decay( _wso.account_subsidy_witness_rd.decay_params, median_available_witness_account_subsidies, 1 );
```

Comment:
This is not what I had envisioned at all for witness subsidy restrictions, but if I am reading this correctly, then this will attempt to distribute account subsidy claims evenly across all witnesses.